src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -499,9 +499,10 @@ def test_generate_schedule_dummy_subroutine(parser):
     schedule = processor.generate_schedule("dummy_code", ast)
     assert isinstance(schedule, KernelSchedule)
 
-    # Test argument intent is inferred when not available in the declaration
+    # Test that argument intent is left as UNKNOWN when not available in the
```

Comment:
The comment and assert is fine but it seems this check is (and was) misplaced as generate_schedule is not dealing directly with the declarations.

This assert will be better placed in the "test_process_declarations_intent".

Reply:
Yes, I agree - done.